packages/agents/src/AgentConfig.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  ExecuteAgentFunctionCalled,
+  Timeout, Scripts
+} from "@evo-ninja/agent-utils";
+import { AgentFunctionBase } from "./AgentFunctionBase";
+import { ReadVariableFunction } from "./functions/ReadVariable";
+import { OnGoalAchievedFunction } from "./functions/OnGoalAchieved";
+import { OnGoalFailedFunction } from "./functions/OnGoalFailed";
+import { AgentPrompts } from "./AgentPrompts";
+
+export class AgentConfig<TRunArgs> {
+  shouldTerminate: (functionCalled: ExecuteAgentFunctionCalled) => boolean;
+  prompts: AgentPrompts<TRunArgs>;
+
+  constructor(
+    promptFactory: (
+      onGoalAchievedFn: AgentFunctionBase<any>,
+      onGoalFailedFn: AgentFunctionBase<any>
+    ) => AgentPrompts<TRunArgs>,
+    readonly functions: AgentFunctionBase<unknown>[],
```

Comment:
I'm confused by why this is working without having `public ` before each argument you want to be a property on the class.

Reply:
I think the readonly makes it a property (cause why would an arg be readonly).
Nonetheless, I'll change it to public to avoid ambiguity